lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2127,14 +2133,26 @@ listStakeKeys lookupStakeRef ctx (ApiT wid) = do
                 <$> liftIO . runExceptT $ W.readRewardAccount @_ @s @k @n wrk wid
             ourApiDelegation <- liftIO $ toApiWalletDelegation (meta ^. #delegation)
                 (unsafeExtendSafeZone (timeInterpreter $ ctx ^. networkLayer))
+
+            -- With ADP-808 we will be able to have many keys. Currently there's
+            -- only one.
             let ourKeys = case mourAccount of
                     Just acc -> [(acc, 0, ourApiDelegation)]
                     Nothing -> []
 
+            -- This uses `getAccountBalance` under the hood. This should give us
+            -- automatic batching of requests, but at the expense of some
+            -- results potentially being zero until listing stake keys once
```

Comment:
I had forgotten `getAccountBalance` worked like this just before. 

It seems like something that should be fixed, but we have the same behaviour for calculating the wallet balance — we only include the reward balance if we have fetched them before, and afterwards we refetch on every tip-change.

Reply:
Should be fixed with 18b36d071eab781397ae2b2a82a6bd25b4df602d